File: Generating/factory/python/MarketFactory.py
from abc import ABC, abstractmethod

class MarketFactory(ABC):
  def printCoinInfo(self):
    coin = self.getCoinInstance()
    print(
      'coinname: ', str(coin.getName()), ' | ',
      'coin balance: ', str(coin.getBalance()), ' | ',
      'coin network: ', str(coin.getNetwork()), ' | '
    )
    pass

  @abstractmethod
  def getCoinInstance():
    pass

File: Generating/factory/cpp/MarketFactory.cpp
#ifndef factory_instance
#define factory_instance

#include <iostream>
#include "CoinInstance.hpp"

using namespace std;

class MarketFactory
{
  public:
    void printCoinInformation() {
      CoinInstance * coin = this->getCoinInstance();

      cout << "Blockchain: " << coin->getName() << endl;
      cout << "Network:    " << coin->getNetwork() << endl;
      cout << "Balance:    " << coin->getBalance() << endl;
    };

    virtual CoinInstance * getCoinInstance() = 0;
};

#endif